openshift/dynamic/client.py

```diff
@@ -126,7 +126,7 @@ def patch(self, resource, body=None, name=None, namespace=None, **kwargs):
         if resource.namespaced:
             namespace = self.ensure_namespace(resource, namespace, body)
 
-        content_type = kwargs.pop('content_type', 'application/strategic-merge-patch+json')
+        content_type = kwargs.pop('content_type', 'application/json-patch+json')
```

Comment:
I think just changing it would alter the behavior of a lot of existing code that expects the strategic-merge to be used by default, is there a way we can more intelligently determine the proper content-type to send?

Reply:
I think this is intelligent enough because we are performing a patch. In particular, for the intelligence, it already exists downstream. As mentioned: https://github.com/openshift/openshift-restclient-python/issues/334 intelligence already exists in the Kubernetes client to upgrade to a strategic merge when the payload is not a list. https://github.com/kubernetes-client/python/blob/3739db034d34b035bc8141740e9073c84ab192c0/kubernetes/client/rest.py#L156

Reply:
Ah I see, sorry I missed that

Reply:
No problem. This only works since it's patch. Technically I feel like there's a bug in Kubernetes client.
Notice how it defaults to `application/json`: https://github.com/kubernetes-client/python/blob/3739db034d34b035bc8141740e9073c84ab192c0/kubernetes/client/rest.py#L146
Then though, it checks for `json` https://github.com/kubernetes-client/python/blob/3739db034d34b035bc8141740e9073c84ab192c0/kubernetes/client/rest.py#L154
Then it checks for `patch` https://github.com/kubernetes-client/python/blob/3739db034d34b035bc8141740e9073c84ab192c0/kubernetes/client/rest.py#L155
If they're going to go through the trouble of defaulting to `application/json` yet check for `json`, shouldn't they then, if the method is PATCH, default to `application/json-patch+json`? Especially because it checks for that as well.

I guess I'd hope the downstream Kubernetes client would either presume the caller sends what should be sent and doesn't assume or it should handle all cases by itself via its defaults + inferences from other data.